game/src/ungap/route/mod.rs

```diff
@@ -148,21 +181,21 @@ impl State<App> for RoutePlanner {
             });
         }
 
-        let mut focused_on_alt_route = false;
-        for r in &mut self.alt_routes {
-            r.event(ctx);
-            focused_on_alt_route |= r.has_focus();
-            if r.has_focus() && ctx.normal_left_click() {
+        let world_outcome_for_waypoints = match self.world.event(ctx) {
+            WorldOutcome::ClickedObject(ID::AltRoute(idx)) => {
                 // Switch routes
-                app.session.routing_preferences = r.results.preferences;
+                app.session.routing_preferences = self.alt_routes[idx].preferences;
                 self.recalculate_routes(ctx, app);
-                self.update_input_panel(ctx, app);
                 return Transition::Keep;
             }
-        }
+            x => x.map_id(|id| match id {
```

Comment:
If the "composite" world gives us an outcome not related to routes, it must be related to waypoints, so we transform the IDs here.